tutorials/grid_coordinates.py

```diff
@@ -0,0 +1,346 @@
+"""
+.. _grid_coordinates:
+
+Gridding coordinates
+====================
+
+Grid coordinates in Verde are used to create points on a regularly spaced grid
+that are then used in the spline method to interpolate between sample data
+points. The grid can be specified either by the number of points in each
+dimension (the *shape*) or by the grid node spacing.
+
+Creating regular grids in Verde is done using the :func:`verde.grid_coordinates`
+function. It creates a set of regularly spaced points in both the west-east and
+south-north directions, i.e. a two-dimensional spatial grid.
+
+First let's create a region that is 1000 units west-east and 1000 units south-
+north, and we will set an initial spacing to 100 units.
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+import verde as vd
+from matplotlib.patches import Rectangle
+
+# create the bounds for the region and set the spacing for the grid points
+spacing = 100
+west, east, south, north = 0, 1000, 0, 1000
+region = (west, east, south, north)
+
+# create the grid
+square_region_east, square_region_north = vd.grid_coordinates(
+    region=region, spacing=spacing
+)
+
+########################################################################################
+# We can check the dimensions of the grid to confirm that it is 11x11
+
+print(square_region_east.shape, square_region_north.shape)
+
+########################################################################################
+# Now let's plot the grid to visualize the location of the grid points
+
+plt.figure(figsize=(6, 6))
+currentAxis = plt.gca()
+currentAxis.add_patch(
+    plt.Rectangle((west, south), east, north, fill=None, label="Region Bounds")
+)
+
+plt.vlines(
+    square_region_east[0],
+    ymin=min(square_region_north[0]),
+    ymax=max(square_region_north[-1]),
+    linestyles="dotted",
+    zorder=0,
+)
+plt.hlines(
+    square_region_north[:, 1],
+    xmin=min(square_region_east[0]),
+    xmax=max(square_region_east[-1]),
+    linestyles="dotted",
+    zorder=0,
+)
+
+plt.scatter(
+    square_region_east,
+    square_region_north,
+    label="Square Region Grid Nodes",
+    marker=".",
+    color="black",
+    s=100,
+)
+
+plt.ylim(-50, 1050)
+plt.xlim(-50, 1050)
+plt.xlabel("Easting")
+plt.ylabel("Northing")
+plt.legend(loc="upper center", bbox_to_anchor=(0.5, 1.15))
+plt.show()

```

Comment:
These plots are great! Would it make sense to have this code in a function since it's repeated below? It could something like:

```python
def plot_grid(ax, coordinates, linestyles="dotted", region=None, pad=50, **kwargs):
    "Plot the grid coordinates as dots and lines."
    if region is not None:
        west, east, south, north = region
        ax.add_patch(
            plt.Rectangle((west, south), east, north, fill=None, label="Region Bounds")
        )
    data_region = vd.get_region(coordinates)
    ax.vlines(
        coordinates[0][0],
        ymin=data_region[2],
        ymax=data_region[3],
        linestyles=linestyles,
        zorder=0,
    )
    ax.hlines(
        coordinates[1][:, 1],
        xmin=data_region[0],
        xmax=data_region[1],
        linestyles=linestyles,
        zorder=0,
    )
    ax.scatter(*coordinates, **kwargs)
    if pad:
        padded = vd.pad_region(region, pad=pad)
        plt.xlim(padded[:2])
        plt.ylim(padded[2:])
```

This way the plotting can be simplified to this later on:

```python

plt.figure(figsize=(6, 6))
plot_grid(
    ax=plt.subplot(111), 
    coordinates=(square_region_east, square_region_north), 
    region=region,
    label="Square Region Grid Nodes",
    marker=".",
    color="black",
    s=100,
)
plt.xlabel("Easting")
plt.ylabel("Northing")
plt.legend(loc="upper center", bbox_to_anchor=(0.5, 1.15))
plt.show()
```